src/components/JobReorderModal.vue

```diff
@@ -0,0 +1,187 @@
+      failedJobs: [] as any,
+      successJobs: [] as any,
+      jobs: (this as any).jobsForReorder,
+      seqBeforeReorder: JSON.parse(JSON.stringify((this as any).jobsForReorder))
+    }
+  },
+  computed: {
+    ...mapGetters({
+      userProfile: "user/getUserProfile"
+    }),
+  },
+  props: ["jobsForReorder"],
+  methods: {
+    timeTillJob (time: any) {
+      const timeDiff = DateTime.fromMillis(time).diff(DateTime.local());
+      return DateTime.local().plus(timeDiff).toRelative();
+    },
+    closeModal(isJobsUpdated = false) {
+      modalController.dismiss({ isJobsUpdated });
+    },
+    getTime (time: any) {
+      return DateTime.fromMillis(time).toLocaleString(DateTime.TIME_SIMPLE);
+    },
+    findJobDiff(previousSeq: any, updatedSeq: any) {
+      // finding the diff using array element position
+      const diffSeq: any = Object.keys(previousSeq).reduce((diff, key) => {
+        if (updatedSeq[key].jobId === previousSeq[key].jobId) return diff
+        return {
+          ...diff,
+          [key]: updatedSeq[key]
+        }
+      }, {})
+      return diffSeq;
+    },
+    doReorder(event: CustomEvent) {
+      // making the item reorder action as complete and storing the updated order in jobs
+      this.jobs = event.detail.complete(JSON.parse(JSON.stringify(this.jobs)));
+    },
+    async save() {
+      let diffSeq = this.findJobDiff(this.seqBeforeReorder, this.jobs)
+
+      // if there are no jobs to update then closing the modal and displaying a toast
+      if(!diffSeq.length) {
+        showToast(translate('No jobs to update'))
+        this.closeModal();
+        return;
+      }
+
+      const updatedRunTime = this.seqBeforeReorder.map((job: any) => job.runTime)
+      Object.keys(diffSeq).map((key: any) => {
+        diffSeq[key].runTime = updatedRunTime[key]
+      })
+      diffSeq = Object.keys(diffSeq).map((key) => diffSeq[key])
+      this.updatedJobsOrder = diffSeq
+
+      this.failedJobs = []
+      this.successJobs = []
+      await Promise.allSettled(this.updatedJobsOrder.map(async (job: any) => {
+        const payload = {
+          'jobId': job.jobId,
+          'systemJobEnumId': job.systemJobEnumId,
+          'recurrenceTimeZone': this.userProfile.userTimeZone,
+          'statusId': "SERVICE_PENDING",
+          'runTime': job.runTime
+        }
+        try {
+          const resp = await JobService.updateJob(payload)
+          if (resp.status == 200 && !hasError(resp) && resp.data.successMessage) {
+            // if the job succeded when updating then adding the jobId to the successJobs array
+            this.successJobs.push(job.jobId)
+            showToast(translate('Jobs sequence updated successfully'))
+          } else {
+            // if the job failed when updating then adding the jobId to the failedJobs array
+            this.failedJobs.push(job.jobId)
+            logger.error('Failed to update some jobs')
+            showToast(translate('Failed to update some jobs'))
```

Comment:
I think this should be done once after all the iterations, if all success we show success message else the error one

Reply:
Yes Sir